### PR TITLE
[autosubmit] Add GitHub app authentication

### PR DESCRIPTION
**Pull request by CaseyHillers**
Fixes https://github.com/flutter/flutter/issues/99435

### CODE REVIEW

**Review by keyonghan — APPROVED** [review]
LGTM with nits.

File: auto_submit/lib/core/providers.dart [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// Copyright 2022 The Flutter Authors. All rights reserved.
```
[the remark this answers is not in the document]
Comment (CaseyHillers) [reply auto_submit/lib/core/providers.dart]: SG. In the future, I think we'll want to rename this into core as it's the reusable parts of the microservices

File: auto_submit/lib/service/config.dart [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+class Config {
+  const Config({
+    required this.cacheProvider,
+    this.httpProvider = Providers.freshHttpClient,
+  });
+
+  // List of environment variable keys related to the Github app authentication.
+  static const String kGithubKey = 'GITHUB_KEY';
+  static const String kGithubAppId = 'GITHUB_APP_ID';
```
[the remark this answers is not in the document]
Comment (CaseyHillers) [reply auto_submit/lib/service/config.dart:26]: Not yet, but i'll publish them after this is working (I want to validate the exception is thrown)

File: auto_submit/lib/service/config.dart [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  final CacheProvider cacheProvider;
+  final HttpProvider httpProvider;
+
+  Cache get cache => Cache(cacheProvider).withPrefix('config').withCodec(utf8);
+
+  Future<GitHub> createGithubClient() async {
+    // GitHub's secondary rate limits are run into very frequently when making auth tokens.
+    final String token = await cache['githubToken'].get(
+      _generateGithubToken,
+      const Duration(minutes: 8),
```
[the remark this answers is not in the document]
Comment (CaseyHillers) [reply auto_submit/lib/service/config.dart:39]: Added

**Pull request merged**